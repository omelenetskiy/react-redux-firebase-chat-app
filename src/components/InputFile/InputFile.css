.attachment__top {
  width: 100%;
  display: inline-flex; }
  .attachment__top .attachment {
    padding: 4px 10px;
    margin: 5px;
    border-radius: 4px;
    border: 1px solid #000000;
    display: inline-block;
    color: #ffffff;
    background-color: #282c34;
    display: inline-flex;
    align-items: center;
    flex-direction: row;
    flex-wrap: nowrap; }
    .attachment__top .attachment span {
      font-size: 10px;
      line-height: 1; }
    .attachment__top .attachment .close {
      color: #ffffff;
      padding: 0 0 0 10px; }
