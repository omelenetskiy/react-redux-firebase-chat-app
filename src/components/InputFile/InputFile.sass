.attachment__top
    width: 100%
    display: inline-flex
    .attachment
        padding: 4px 10px
        margin: 5px
        border-radius: 4px
        border: 1px solid #000000
        display: inline-block
        color: #ffffff
        background-color: rgba(40, 44, 52, 1)
        display: inline-flex
        align-items: center
        flex-direction: row
        flex-wrap: nowrap
        span 
            font-size: 10px
            line-height: 1
        .close
            color: #ffffff
            padding: 0 0 0 10px