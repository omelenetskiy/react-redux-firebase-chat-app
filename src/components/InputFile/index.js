import React from 'react';
import PropTypes from 'prop-types';
import './InputFile.sass';

const InputFile = ({ fileName, close }) => (
  <div className="attachment__top">
    <div className="attachment">
      <span>
        {fileName}
      </span>
      <button type="button" className="close" onClick={() => close()}>
        <i className="far fa-times-circle" />
      </button>
    </div>
  </div>
);

export default InputFile;

InputFile.propTypes = {
  fileName: PropTypes.string.isRequired,
  close: PropTypes.func.isRequired,
};
