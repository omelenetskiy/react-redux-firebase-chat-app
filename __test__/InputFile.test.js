import React from 'react';
import InputFile from '../src/components/InputFile';
import { shallow, render } from 'enzyme';

describe('<InputFile />', () => {
	it('should render a Loading', () => {
		const wrapper = shallow(
			<InputFile fileName="File" close={() => closeFile()} />
		);
		expect(wrapper).toMatchSnapshot();
	});
	it('should render a document name', () => {
		const wrapper = render(
			<InputFile fileName="File" close={() => closeFile()} />
		);
		expect(wrapper.find('span').text()).toEqual('File');
	});
});
