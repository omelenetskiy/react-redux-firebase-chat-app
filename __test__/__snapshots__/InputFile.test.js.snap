// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputFile /> should render a Loading 1`] = `
<div
  className="attachment__top"
>
  <div
    className="attachment"
  >
    <span>
      File
    </span>
    <button
      className="close"
      onClick={[Function]}
    >
      <i
        className="far fa-times-circle"
      />
    </button>
  </div>
</div>
`;
